src/globalComponents/apiCalls/getUserAPICall.js

```diff
@@ -2,27 +2,30 @@
 
 import getLoginToken from '../getLoginToken.js';
 import User from '../typesAndClasses/User.js';
+import { getInventory } from '../GetInventoryInfo.js';
+import { toast } from 'react-toastify';
+import getErrorFromObject from '../getErrorFromObject.js';
 
 //gets the user when passed a token stored as the login token
 function getUserAPICall(onLoad:(user: User) => void): void {
```

Comment:
You still have this issue here. According to the types, user should never be null. But you do a bunch of user==null checks in this diff, which is very confusing. Either you are making assumptions about the type and those assumptions aren't true, or you have redundant checks, which is confusing.